things/api/http/endpoint_test.go

```diff
@@ -280,6 +290,126 @@ func TestUpdateThing(t *testing.T) {
 	}
 }
 
+func TestUpdateKey(t *testing.T) {
+	svc := newService(map[string]string{token: email})
+	ts := newServer(svc)
+	defer ts.Close()
+
+	th := thing
+	th.Key = "key"
+	sth, _ := svc.AddThing(token, th)
+
+	sth.Key = "new-key"
+	data := toJSON(sth)
+
+	sth.Key = "key"
+	dummyData := toJSON(sth)
+
+	cases := []struct {
+		desc        string
+		req         string
+		id          string
+		contentType string
+		auth        string
+		status      int
+	}{
+		{
+			desc:        "update key for an existing thing",
+			req:         data,
+			id:          sth.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusOK,
+		},
+		{
+			desc:        "update thing with conflicting key",
+			req:         data,
+			id:          sth.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusConflict,
```

Comment:
I think that http [status 409](https://stackoverflow.com/a/20215807) is not appropriate for this. 
I think that 422 Unprocessable entity would be more appropriate. This error means that the server understood the request but was unable to process it because of an error.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
https://stackoverflow.com/a/20215807

Reply:
@chombium I'm open for suggestions, but can you explain to me why it's not appropriate? Key should be unique and if it's not for me it makes sense that server returns `409 Conflict`. If you think that in this way one user can find a key of a thing that he shouldn't have access to, well, I'm not sure that this is a problem, because he still cannot see other users channels and also doesn't now to which thing specified key belongs to.

Reply:
I missed the first reference sorry, everything is clear.

Reply:
But [here](https://httpstatuses.com/409) it says that `409` status should be used in such situations. Not sure what to do.

Reply:
OK, final decision, I'll use `422`. `409` is used when request is in a conflict with the current state of a target resource which is not the case here. Thanks @chombium!

Reply:
Fixed.

Reply:
409 would be a good response if two users read the same data record, the first updates it and then the second tries to update it. The second update should get a 409 because the data has changed and he/she has tried to update older version of the data